kbp/config/config_v1_test.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package config
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestConfigV1Default(t *testing.T) {
+	config := DefaultV1()
+	read, list, err := config.GetPermissionsForAnonymous("/")
+	require.NoError(t, err)
+	require.True(t, read)
+	require.False(t, list)
+}
+
+func TestConfigV1Invalid(t *testing.T) {
+	_, _, err := (&V1{
+		Common: Common{
+			Version: Version1Str,
+		},
+		DefaultACL: AccessControlV1{
+			WhitelistAdditionalPermissions: map[string]string{
+				"alice": "read",
+			},
+		},
+	}).GetPermissionsForAnonymous("/")
+	require.Error(t, err)
+	require.IsType(t, ErrUndefinedUsername{}, err)
+
+	_, _, err = (&V1{
+		Common: Common{
+			Version: Version1Str,
+		},
+		ACLs: map[string]AccessControlV1{
+			"/foo": AccessControlV1{
+				AnonymousPermissions: "",
+			},
+			"/foo/../foo": AccessControlV1{
+				AnonymousPermissions: "read",
+			},
+		},
+	}).GetPermissionsForAnonymous("/")
+	require.Error(t, err)
+	require.IsType(t, ErrDuplicateAccessControlPath{}, err)
+
+	_, _, err = (&V1{
+		Common: Common{
+			Version: Version1Str,
+		},
+		DefaultACL: AccessControlV1{
+			AnonymousPermissions: "huh?",
+		},
+	}).GetPermissionsForAnonymous("/")
+	require.Error(t, err)
+	require.IsType(t, ErrInvalidPermissions{}, err)
+}
+
+func TestConfigV1Full(t *testing.T) {
+	config := V1{
+		Common: Common{
+			Version: Version1Str,
+		},
+		Users: map[string]UserV1{
+			"alice": UserV1{
+				PasswordSalt:      "salt",
+				PasswordSHA512Hex: MakeSaltedSHA512PasswordHex("12345", "salt"),
+			},
+			"bob": UserV1{
+				PasswordSalt:      "tlas",
+				PasswordSHA512Hex: MakeSaltedSHA512PasswordHex("54321", "tlas"),
+			},
+		},
+		DefaultACL: AccessControlV1{
+			AnonymousPermissions: "read",
+		},
+		ACLs: map[string]AccessControlV1{
+			"/alice-and-bob": AccessControlV1{
+				AnonymousPermissions: "",
```

Comment:
Can you make one of these not specify an explicit `AnonymousPermissions: ""`?  Should behave the same way, but I wouldn't expect users to actually do this most of the time, so it's a good thing to check.